spec/inputs/elasticsearch_esql_spec.rb

```diff
@@ -0,0 +1,95 @@
+# encoding: utf-8
+require "logstash/devutils/rspec/spec_helper"
+require "logstash/inputs/elasticsearch"
+require "elasticsearch"
+
+describe LogStash::Inputs::Elasticsearch::Esql do
+  let(:client) { instance_double(Elasticsearch::Client) }
+  let(:esql_client) { double("esql-client") }
+  let(:plugin) { instance_double(LogStash::Inputs::Elasticsearch, params: plugin_config) }
+  let(:plugin_config) do
+    {
+      "query" => "FROM test-index | STATS count() BY field",
+      "retries" => 3
+    }
+  end
+  let(:esql_executor) { described_class.new(client, plugin) }
+
+  describe "when initializes" do
+    it "sets up the ESQL client with correct parameters" do
+      expect(esql_executor.instance_variable_get(:@query)).to eq(plugin_config["query"])
+      expect(esql_executor.instance_variable_get(:@retries)).to eq(plugin_config["retries"])
+    end
+  end
+
+  describe "when faces error while retrying" do
+    it "retries the given block the specified number of times" do
+      attempts = 0
+      result = esql_executor.retryable("Test Job") do
+        attempts += 1
+        raise StandardError if attempts < 3
+        "success"
+      end
+      expect(attempts).to eq(3)
+      expect(result).to eq("success")
+    end
+
+    it "returns false if the block fails all attempts" do
+      result = esql_executor.retryable("Test Job") do
+        raise StandardError
+      end
+      expect(result).to eq(false)
+    end
+  end
+
+  describe "when executing chain of processes" do
```

Comment:
minor nitpick for rspec hygiene, typically for "when .." you use `context`:

```ruby
context "when execution chain of processes" do
end
```

versus when describing a particular feature or setting, using `describe`, typically also coupled with `context`:

```ruby
describe "response handling" do
  # set up overall request/response
  context "when response does not contain a warning header" do
    # set up individual case of not containing a warning header
    it "does not log a warning" do
      # test the individual case
    end
  end
end

Reply:
Thanks for catching this up. I made them maned correctly in `elasticsearch_spec.rb` unit tests but somehow missed here. I have corrected them.